packages/react/src/meter/root/useMeterRoot.ts

```diff
@@ -0,0 +1,179 @@
+  let isOptimal = false;
+
+  if (min <= optimum && optimum <= low) {
+    isOptimal = segment === 'low';
+  } else if (high <= optimum && optimum <= max) {
+    isOptimal = segment === 'high';
+  }
+
+  const getRootProps: useMeterRoot.ReturnValue['getRootProps'] = React.useCallback(
+    (externalProps = {}) =>
+      mergeReactProps<'div'>(externalProps, {
+        'aria-label': getAriaLabel ? getAriaLabel(value) : ariaLabel,
+        'aria-labelledby': ariaLabelledby,
+        'aria-valuemax': max,
+        'aria-valuemin': min,
+        'aria-valuenow': percentageValue / 100,
+        'aria-valuetext': getAriaValueText
+          ? getAriaValueText(value)
+          : (ariaValuetext ?? `${percentageValue}%`),
+        role: 'meter',
+      }),
+    [
+      ariaLabel,
+      ariaLabelledby,
+      ariaValuetext,
+      getAriaLabel,
+      getAriaValueText,
+      max,
+      min,
+      value,
+      percentageValue,
+    ],
+  );
+
+  return {
+    getRootProps,
+    max,
+    min,
+    value,
+    percentageValue,
+    segment,
+    isOptimal,
+  };
+}
+
+namespace useMeterRoot {
+  export type Segment = 'low' | 'medium' | 'high';
+
+  export interface Parameters {
+    /**
+     * The label for the Indicator component.
+     */
+    'aria-label'?: string;
+    /**
+     * An id or space-separated list of ids of elements that label the Indicator component.
+     */
+    'aria-labelledby'?: string;
+    /**
+     * A string value that provides a human-readable text alternative for the current value of the meter indicator.
+     */
+    'aria-valuetext'?: string;
+    /**
+     * Accepts a function which returns a string value that provides an accessible name for the Indicator component
+     * @param {number} value The component's value
+     * @returns {string}
+     */
+    getAriaLabel?: (value: number) => string;
+    /**
+     * Accepts a function which returns a string value that provides a human-readable text alternative for the current value of the meter indicator.
+     * @param {number} value The component's value to format
+     * @returns {string}
+     */
+    getAriaValueText?: (value: number) => string;
+    /**
+     * Sets the lower boundary of the high end of the numeric range represented by the component.
+     * If unspecified, or greater than `max`, it will fall back to `max`.
+     * @default 100
+     */
+    high?: number;
+    /**
+     * Sets the upper boundary of the low end of the numeric range represented by the component.
+     * If unspecified, or less than `min`, it will fall back to `min`.
+     * @default 0
+     */
+    low?: number;
+    /**
+     * The maximum value
+     * @default 100
+     */
+    max?: number;
+    /**
+     * The minimum value
+     * @default 0
+     */
+    min?: number;
+    /**
+     * Indicates the optimal point in the numeric range represented by the component.
+     * If unspecified, it will fall back to the midpoint between `min` and `max`.
```

Comment:
This API feels a bit weird to me, but I understand it reflects the native `<meter>` element.
I think I wouldn't set the default value for `optimum` if not provided and not set the `data-optimum` attribute in this case.

Reply:
Sounds good, it does seem a bit arbitrary anyway